test/Test/Database/LSMTree/Internal/Run/Index/Compact.hs

```diff
@@ -246,3 +368,66 @@ eqMapProp m1 m2 = conjoin . Map.elems $
       printf "Key-value pair only on the right, (k, y) = (%s, %s)" (show k) (show y)
     both k x y = flip counterexample (property (x == y)) $
       printf "Mismatch between x and y, (k, x, y) = (%s, %s, %s)" (show k) (show x) (show y)
+
+showBytes :: [Word8] -> [String]
+showBytes = map (unwords . map (foldMap showByte) . chunksOf 4) . chunksOf 8
+
+showByte :: Word8 -> String
+showByte b = let str = showHex b "" in replicate (2 - length str) '0' <> str
+
+buildBytes :: BB.Builder -> [Word8]
+buildBytes = LBS.unpack . BB.toLazyByteString
+
+word32toBytesLE :: Word32 -> [Word8]
+word32toBytesLE = take 4 . map fromIntegral . iterate (`div` 256)
+
+data Chunks = Chunks [Chunk] FinalChunk
+  deriving (Show)
+
+-- | The only invariants we make sure to uphold is that the length of the
+-- vectors match 'fcRangeFinderPrecision' and 'fcNumPages' respectively,
+-- as this is required for correct deserialisation.
+chunksInvariant :: Chunks -> Bool
+chunksInvariant (Chunks chunks FinalChunk {..}) =
```

Comment:
We should also put a comment here that says these invariants are sufficient for serialisation/deserialisation only. `fromChunks` on a `Chunks` that is generated with this invariant would not lead to a valid `CompactIndex`

Reply:
I see that this is commented elsewhere too, but repeating it doesn't hurt